avm/ptn/app/iaas-vm-cosmosdb-tier4/main.bicep

```
@@ -0,0 +1,828 @@
+metadata name = 'IaaS VM with CosmosDB Tier 4'
+metadata description = 'Creates an IaaS VM with CosmosDB Tier 4 resiliency configuration.'
+
+@description('Required. Name of the solution which is used to generate unique resource names.')
+param name string
+
+@description('General. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('General. Tags for all resources. Should include standard tags like Environment, Owner, CostCenter, etc.')
+@metadata({
+  example: {
+    Environment: 'Production'
+    Owner: 'TeamName'
+    CostCenter: 'IT'
+    Application: 'MyApp'
+  }
+})
+param tags object?
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+// Network security group parameters
+@description('Security. Network security group rules for the application subnet.')
+param applicationNsgRules array = [
```

Also, can be typed via the exported type for the same property of the referenced module
```suggestion
import { securityRuleType } from 'br/public:avm/res/network/network-security-group:0.5.1'
param applicationNsgRules securityRuleType[] = [
```